File: lib/spec/bowlingSpec.js
describe('Game', function() {

  var game

  beforeEach(function() {
    game = new Game
  });

  it('starts with a score of 0',function() {
    expect(game.score).toBe(0)
  });

  it('tracks player rolls',function() {
    game.roll(5);
    game.roll(1);
    game.roll(3);
    expect(game.rolls[1]).toEqual([5,1]);
    expect(game.rolls[2]).toEqual([3])
  });

  it('tracks current frame',function() {
    game.roll(4);
    expect(game.currentFrame).toEqual(1);
    game.roll(4);
    expect(game.currentFrame).toEqual(2);
    game.roll(4);
    expect(game.currentFrame).toEqual(2);
    game.roll(4);
    expect(game.currentFrame).toEqual(3);
    game.roll(10);
    expect(game.currentFrame).toEqual(4)
  });

  it('frames account for strikes',function() {
    game.roll(10);
    game.roll(10);
    game.roll(10);
    expect(parseInt(game.rolls[1])).toEqual(30)
    expect(parseInt(game.rolls[2])).toEqual(20)
    expect(parseInt(game.rolls[3])).toEqual(10)
  });

  it('frames account for spares',function() {
    game.roll(5);
    game.roll(5);
    game.roll(10);
    game.roll(2);
    game.roll(3);
    expect(game.score).toEqual(40)
  });

  it('can score a perfect game',function() {
    for (i = 0 ; i < 12 ; i++) {
      game.roll(10);
    };
    expect(game.score).toEqual(300)
  });

  it('spares and strikes add up correctly',function() {
    game.roll(10);
    game.roll(2);
    game.roll(8);
    game.roll(3);
    game.roll(3);
    expect(game.score).toEqual(39)
  });

  it('can take a full game of spares',function() {
    for (i = 0 ; i < 21 ; i++) {
      game.roll(5)
    }
    expect(game.score).toEqual(150)
  });
});

// var Game = new Game;
// module.exports = Game;

File: ruby/spec/bowling_spec.rb
require_relative '../bowling'

describe 'Bowling game logic' do
  before do
    @game = Bowling.new
  end

  it 'can roll a gutter game' do
    20.times{@game.roll(0)}
    expect(@game.score).to eq 0
  end

  it 'can roll all 1s' do
    20.times{@game.roll(1)}
    expect(@game.score).to eq 20
  end

  it 'can roll a spare' do
    @game.roll(5)
    @game.roll(5)
    @game.roll(3)
    17.times do
      @game.roll(0)
    end
    expect(@game.score).to eq 16
  end

  it 'can roll a strike' do
    @game.roll(10)
    @game.roll(3)
    @game.roll(4)
    16.times do
      @game.roll(0)
    end
    expect(@game.score).to eq 24
  end

  it 'can roll a perfect game' do
    12.times{@game.roll(10)}
    expect(@game.score).to eq 300
  end
end